pkg/kubelet/cm/dra/plugin/client.go

```diff
@@ -24,130 +24,183 @@ import (
 	"google.golang.org/grpc"
 	grpccodes "google.golang.org/grpc/codes"
 	grpcstatus "google.golang.org/grpc/status"
-	"k8s.io/klog/v2"
 
+	"k8s.io/klog/v2"
 	drapbv1alpha2 "k8s.io/kubelet/pkg/apis/dra/v1alpha2"
 	drapb "k8s.io/kubelet/pkg/apis/dra/v1alpha3"
 )
 
 const PluginClientTimeout = 45 * time.Second
 
-// draPluginClient encapsulates all dra plugin methods.
-type draPluginClient struct {
-	pluginName string
-	plugin     *Plugin
-}
+type (
+	nodePrepareResourceHandler   func(context.Context, *grpc.ClientConn, *plugin, *drapb.NodePrepareResourcesRequest) (*drapb.NodePrepareResourcesResponse, error)
+	nodeUnprepareResourceHandler func(context.Context, *grpc.ClientConn, *plugin, *drapb.NodeUnprepareResourcesRequest) (*drapb.NodeUnprepareResourcesResponse, error)
+)
+
+var (
+	nodePrepareResourceHandlers = map[string]nodePrepareResourceHandler{
+		v1alpha2Version: v1alpha2NodePrepareResouce,
+		v1alpha3Version: v1alpha3NodePrepareResouce,
```

Comment:
Does it make more sense to introduce an interface rather than two standalone function types?

So something like:
```
type nodeResourceManager interface {
	Prepare(context.Context, *grpc.ClientConn, *plugin, *drapb.NodePrepareResourcesRequest) (*drapb.NodePrepareResourcesResponse, error)
	Unprepare(context.Context, *grpc.ClientConn, *plugin, *drapb.NodeUnprepareResourcesRequest) (*drapb.NodeUnprepareResourcesResponse, error)
}

type v1alpha2NodeResourceManager struct {}
type v1alpha3NodeResourceManager struct {}

var nodeResourceManagers = map[string]nodeResourceManager{
	v1alpha2Version: v1alpha2NodeResourceHandler{},
	v1alpha3Version: v1alpha3NodeResourceHandler{},
}

func (h v1alpha2NodeResourceManager) Prepare(...) {...}
func (h v1alpha2NodeResourceManager) Unprepare(...) {...}

func (h v1alpha3NodeResourceManager) Prepare(...) {...}
func (h v1alpha3NodeResourceManager) Unprepare(...) {...}
```

Reply:
imho to me it does not make more sense but if you feel more comfortable I can go this way

Reply:
@klueska done implemented your suggestion